giskard/scanner/scanner.py

```diff
@@ -183,6 +148,64 @@ def _postprocess(self, issues: Sequence[Issue]) -> Sequence[Issue]:
 
         return issues
 
+    def _validate_model_and_dataset(self, model, dataset):
+        if not isinstance(model, BaseModel):
+            raise ValueError(
+                "The model object you provided is not valid. Please wrap it with the `giskard.Model` class. "
+                "See the instructions here: https://docs.giskard.ai/en/latest/guides/wrap_model/index.html"
+            )
+
+        if dataset is not None and not isinstance(dataset, Dataset):
+            raise ValueError(
+                "The dataset object you provided is not valid. Please wrap your dataframe with `giskard.Dataset`. "
+                "You can follow the docs here: https://docs.giskard.ai/en/latest/guides/wrap_dataset/index.html"
+            )
+
+        # Some extra checks
+        if model.is_text_generation:
+            if not model.meta.name or not model.meta.description:
+                raise ValueError(
+                    "LLM models must have a name and a description. Please make sure the `name` and `description` "
+                    "parameters of your model are not empty. We will use them to configure our detectors."
+                )
+            if not model.meta.feature_names:
+                raise ValueError(
+                    "LLM models must specify their input variables to be analyzed. "
+                    "Please make sure to set the `feature_names` parameter when wrapping your model."
+                )
```

Comment:
let's remove these in favor of https://github.com/Giskard-AI/giskard/pull/1525

Reply:
out of scope for this PR, validation should be handled in #1525